src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -1534,6 +1564,205 @@ subroutine init_atm_static(mesh, dims, configs)
+                return
+            end if
+        end if
+
+        !
+        ! Process each tile by removing it from the stack. Determine the closest cell center to each tile
+        ! pixel by using a KD search and pass the pixel value and the closest cell center id to the
+        ! accumulation routine.
+        !
+        do while (.not. mgr % is_stack_empty())
+            tile => mgr % pop_tile()
+
+            if (tile % is_processed) then
+                cycle
+            end if
+
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do j = supersample_fac_lcl * tile_bdr + 1, supersample_fac_lcl * (tile_ny + tile_bdr), subsample_fac
+                do i = supersample_fac_lcl * tile_bdr + 1, supersample_fac_lcl * (tile_nx + tile_bdr), subsample_fac
+
+                    ! Supersample coordinates
+                    ii = (i - 1) / supersample_fac_lcl + 1
+                    jj = (j - 1) / supersample_fac_lcl + 1
+
+                    call mgr % tile_to_latlon(tile, j, i, lat, lon, supersample_fac_lcl)
+                    call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat, lon)
+                    call mpas_kd_search(kdtree, (/xPixel, yPixel, zPixel/), res)
+
+                    !
+                    ! For outermost boundary cells, extra work is needed to be done to determine if a pixel actually
+                    ! lies within the cell returned by mpas_kd_search
+                    !
+                    if (bdyMaskCell(res % id) == nBdyLayers) then
+                        ! mpas_in_cell could be included in the if statement above, but calling it is expensive and the Fortran
+                        ! standard does require compilers to short circuit compound conditionals.
+                        if (.not. mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                               nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+                            !
+                            ! This pixel lies outside of res % cell and outside of the limited-area region, no further processing is
+                            ! needed
+                            !
+                            cycle
+                        end if
+                    end if
+
+                    !
+                    ! Send the entire pixel column to the accumulation method
+                    !
+                    call accumulation_method(mgr, res % id, int(tile % tile(ii,jj,:), kind=I8KIND))
+
+                    if (res % id <= nCellsSolve) then
+                        all_pixels_mapped_to_halo_cells = .false.
+                    end if
+                end do
+            end do
+
+            tile % is_processed = .true.
+
+            !
+            ! If a single pixel value maps to an owned cell (i.e. <= nCellsSolve) then
```

Comment:
It might be clearer here to say "If at least one pixel" rather than "If a single pixel value", as the latter statement could be interpreted to mean "If exactly one pixel value".

Reply:
I agree and fixed. Thanks for the suggestion.